libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/MockChain.hs

```diff
@@ -113,6 +116,14 @@ instance Show (MockBlock era) where
 instance Reflect era => TotalAda (MockChainState era) where
   totalAda (MockChainState nes _ _) = totalAda nes
 
+deriving instance Generic (MockChainState era)
+
+instance
+  ( Era era,
+    NoThunks (NewEpochState era)
+  ) =>
+  NoThunks (MockChainState era)
```

Comment:
I don't think there is a reason to span over five lines of code something that can fit into one. Longer files are harder to read, we should save that real state if we can.

```suggestion
instance (Era era, NoThunks (NewEpochState era)) => NoThunks (MockChainState era)
```